packages/resource-utils/src/validResourceSegmentPattern.ts

```diff
@@ -40,8 +40,32 @@ const invalidBranchNamePattern = [
   `/$`,
 ].join("|");
 
+// Removed ?,^, [, \, * from invalidBranchNamePattern as they are allowed in fnmatch
+const invalidBranchProtectionNamePattern = [
```

Comment:
Are these all in `invalidBranchNamePattern` too? Can spread this array in there so it's clearer what the differences are